pkg/gce-pd-csi-driver/node.go

```diff
@@ -349,7 +351,36 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 }
 
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID must be provided")
+	}
+	capacityRange := req.GetCapacityRange()
+	reqBytes, err := getRequestCapacity(capacityRange)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume capacity range is invalid: %v", err))
+	}
+
+	volKey, err := common.VolumeIDToKey(volumeID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume volume ID is invalid: %v", err))
+	}
+
+	devicePath, err := ns.getDevicePath(volumeID, "")
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Error when getting device path for %s: %v", volumeID, err))
+	}
+
+	resizer := resizefs.NewResizeFs(ns.Mounter)
```

Comment:
If this is a raw block volume, we should skip trying to resize the filesystem.

Reply:
`resizer.Resize` already checks for `fstype` and skips resize if no `fs` is found (block device).

Reply:
Are we able to unit test this? If not, e2e test?

Reply:
wrote a unit test, it's kind of insane. suggestions welcome lol

Reply:
Thanks for adding a unit test. We should probably add a e2e test anyway, so that we can actually catch potential distro issues too.

Reply:
sure, we can't do `xfs` just yet since we don't bring up our e2e tests with an `xfs` distro but that's a seperate issue. I can add one for the block device case